logr.go

```diff
@@ -387,6 +387,10 @@ func (l Logger) IsZero() bool {
 // that it is meant to log from a context, then the values for those keys from
 // the given context will be added to all log entries.
 func (l Logger) WithContext(ctx context.Context) Logger {
+	if l.sink == discardLogSinkInstance {
```

Comment:
This cannot be `l.sink == discardLogSink{}` (not allowed by Go), therefore I introduced `discardLogSinkInstance`.

Reply:
Could be:

```
if _, ok := l.sink.(discardLogSink); ok {
```

But this is fine too.